lib/mongoid/traversable.rb

```diff
@@ -27,8 +29,7 @@ def discriminator_key=(value)
         end
 
         if value
-          # I don't need to do any sort of validation here because it will be done
-          # on field creation.
+          Mongoid::Fields::Validators::Macro.validate(self, value, {})
```

Comment:
Validating here gives an undesired outcome, in my opinion... If a user sets the discriminator key to a field that already exists, it shouldn't fail, it should do nothing... That's why we have the `if` statement online the 47 of this file... Do you agree @p ?

Reply:
Especially because this is going to be validated when the field is added on line 49... I guess the only edge case is when creating an invalid discriminator key before the child class is instantiated, and then the validation doesn't happen until it does get instantiated...

Alternatively, I can do what I did in `Mongoid` and throw an `InvalidDiscriminatorKey` error on the field name being in `Mongoid.destructive_fields`.